src/uct/ib/base/ib_iface.c

```diff
@@ -177,7 +177,27 @@ int uct_ib_iface_is_reachable(const uct_iface_h tl_iface, const uct_device_addr_
     uint16_t lid;
 
     uct_ib_address_unpack(ib_addr, &lid, &is_global, &gid);
-    return (gid.global.subnet_prefix == iface->gid.global.subnet_prefix);
+
+#if HAVE_DECL_IBV_LINK_LAYER_INFINIBAND
+    if (!(ib_addr->flags & UCT_IB_ADDRESS_FLAG_GID) ) {
```

Comment:
can be done much simpler:
#ifdef HAVE_ ... _ETHERNET
if (link_layer == ETH) {
   return flags & ETH;
} else if (link_layer == IB) 
#endif
{
 return (flags & IB) && (gid.subnet_prefix == iface->subnet_prefix)
}